packages/zoe/src/contractSupport/auctions.js

```diff
@@ -1,64 +1,51 @@
-export const secondPriceLogic = (bidAmountMath, bidOfferHandles, bids) => {
+export const secondPriceLogic = (bidAmountMath, bidSeats) => {
   let highestBid = bidAmountMath.getEmpty();
   let secondHighestBid = bidAmountMath.getEmpty();
-  let highestBidOfferHandle;
-  // eslint-disable-next-line array-callback-return
-  bidOfferHandles.map((offerHandle, i) => {
-    const bid = bids[i];
+  let highestBidSeat;
+
+  bidSeats.forEach(bidSeat => {
+    const bid = bidSeat.getAmountAllocated('Bid', highestBid.brand);
     // If the bid is greater than the highestBid, it's the new highestBid
     if (bidAmountMath.isGTE(bid, highestBid)) {
       secondHighestBid = highestBid;
       highestBid = bid;
-      highestBidOfferHandle = offerHandle;
+      highestBidSeat = bidSeat;
     } else if (bidAmountMath.isGTE(bid, secondHighestBid)) {
       // If the bid is not greater than the highest bid, but is greater
       // than the second highest bid, it is the new second highest bid.
       secondHighestBid = bid;
     }
   });
   return harden({
-    winnerOfferHandle: highestBidOfferHandle,
+    winnerSeat: highestBidSeat,
     winnerBid: highestBid,
     price: secondHighestBid,
   });
 };
 
-export const closeAuction = (
-  zcf,
-  { auctionLogicFn, sellerOfferHandle, allBidHandles },
-) => {
-  const { brandKeywordRecord } = zcf.getInstanceRecord();
-  const bidAmountMath = zcf.getAmountMath(brandKeywordRecord.Ask);
-  const assetAmountMath = zcf.getAmountMath(brandKeywordRecord.Asset);
-
-  // Filter out any inactive bids
-  const { active: activeBidHandles } = zcf.getOfferStatuses(
-    harden(allBidHandles),
-  );
+export const closeAuction = (zcf, auctionLogicFn, sellSeat, bidSeats) => {
+  const {
+    want: { Ask: minBid },
```

Comment:
FYI I filed #1482 for a more extensive cleanup of these auctions. This specific binding to a set of Keywords makes this seem like a private function for `publoicAuction` rather than a reusable public helper.